lightkurve/tests/test_synthetic_data.py

```diff
@@ -0,0 +1,188 @@
+
+
+lacks_BLS = False
+try:
+    from astropy.stats.bls import BoxLeastSquares
+except ImportError:
+    lacks_BLS = True
+
+def test_sine_SFF():
+    """Test the SFF implementation with known signals"""
+    # Retrieve the custom, known signal properties
+    tpf = KeplerTargetPixelFile(filename_synthetic_sine)
+    true_period = np.float(tpf.hdu[3].header['PERIOD'])
+    true_amplitude = np.float(tpf.hdu[3].header['SINE_AMP'])
+
+    # Run the SFF algorithm
+    lc = tpf.to_lightcurve()
+    corrector = SFFCorrector(lc)
+    cor_lc = corrector.correct(tpf.pos_corr2, tpf.pos_corr1,
+                                 niters=4, windows=5, bins=7, restore_trend=True)
+
+    # Verify that we get the period within ~20%
+    pg = cor_lc.to_periodogram(method='lombscargle', minimum_period=1,
+                                 maximum_period=10, oversample_factor=10)
+    ret_period = pg.period_at_max_power.value
+    threshold = 0.2
+    assert ((ret_period > true_period*(1-threshold)) &
+            (ret_period < true_period*(1+threshold)) )
+
+    # Verify that we get the amplitude to within a factor of 2
+    n_cad = len(tpf.time)
+    design_matrix = np.vstack([np.ones(n_cad),
+                              np.sin(2.0*np.pi*cor_lc.time/ret_period),
+                              np.cos(2.0*np.pi*cor_lc.time/ret_period)]).T
+    ATA = np.dot(design_matrix.T,  design_matrix / cor_lc.flux_err[:, None]**2)
+    least_squares_coeffs = np.linalg.solve(ATA, np.dot(design_matrix.T, cor_lc.flux/cor_lc.flux_err**2 ))
+    const, sin_weight, cos_weight = least_squares_coeffs
+
+    fractional_amplitude = (sin_weight**2+cos_weight**2)**(0.5) / const
+    assert ((fractional_amplitude > true_amplitude/2) &
+            (fractional_amplitude < true_amplitude*2) )
+
+@pytest.mark.skipif(lacks_BLS, reason="Astropy BLS requires Python 3")
+def test_transit_SFF():
+    """Test the SFF implementation with known signals"""
+    # Retrieve the custom, known signal properties
+    tpf = KeplerTargetPixelFile(filename_synthetic_transit)
+    true_period = np.float(tpf.hdu[3].header['PERIOD'])
+    true_depth = np.float(tpf.hdu[3].header['RPRS'])**2.0
+
+    # Run the SFF algorithm
+    lc = tpf.to_lightcurve()
+    corrector = SFFCorrector(lc)
+    cor_lc = corrector.correct(tpf.pos_corr2, tpf.pos_corr1,
+                                 niters=4, windows=5, bins=7, restore_trend=False)
+
+    # Verify that we get the transit period within 5%
+    pg = cor_lc.to_periodogram(method='bls', minimum_period=1, maximum_period=9,
+                               frequency_factor=0.05, duration=np.arange(0.1, 0.6, 0.1))
+    ret_period = pg.period_at_max_power.value
+    threshold = 0.05
+    assert ((ret_period > true_period*(1-threshold)) &
+            (ret_period < true_period*(1+threshold)) )
+
+    # Verify that we get the transit depth to within a factor of 50%
+    assert ((pg.depth_at_max_power > true_depth/1.5) &
+            (pg.depth_at_max_power < true_depth*1.5))
```

Comment:
50% seems a bit generous here; we can't claim that our SFF works well if it overfits by 50%.  At minimum let's reduce this to 20%  (which still passes), and ideally, let's plot detrended vs ideal light curve to check that nothing crazy is going on here?

Reply:
Since BLS approximates a true transit shape, I will modify the benchmark from the coarse Rp/Rs (existing), to a mean of the in-transit points, which I will define as anything within, say, two hundred PPM of the deepest part of the transit.